File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/filters/FilterOption.java
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.filters;

import java.util.List;

import nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.entities.PhotoEntity;

/**
 * Created by stephan on 17/05/2017.
 */

public interface FilterOption {
    PhotoEntity getPhoto(int index);

    List<PhotoEntity> getAll();

    long getCount();
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/filters/FilterOption.kt
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.filters

import nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.entities.PhotoEntity

/**
 * Created by stephan on 17/05/2017.
 */

interface FilterOption {

    val count: Int

    fun getPhoto(index: Int): PhotoEntity
}
